### PR TITLE
Extend decorator documentation

### PR DESCRIPTION
**Pull request by fregante**
I just found out that the proposal only supports class methods :(

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -198,7 +198,13 @@ Refer to the [caching strategies](#caching-strategy) section for more informatio
 
 ### mem.decorator(options)
 
-Returns a TypeScript decorator which memoizes the given function.
+Returns a [decorator](https://github.com/tc39/proposal-decorators) to memoize class methods.
+
+Notes:
+
+- Only class methods and getters/setters can be memoized, not regular functions (they aren't part of the proposal);
```
Comment (fregante) [inline readme.md:205]: ?

```suggestion
- Only class methods and accessors can be memoized, not regular functions (they aren't part of the proposal);
```

Comment (fregante) [reply readme.md:205]:
```suggestion
- Only class methods and accessors can be memoized, not regular functions ([they aren't part of the proposal]());
```

Comment (papb) [reply readme.md:205]: I personally prefer `getters/setters`

File: readme.md [hunk 2]
```diff
@@ -198,7 +198,13 @@ Refer to the [caching strategies](#caching-strategy) section for more informatio
...
+- Only [TypeScript’s decorators](https://www.typescriptlang.org/docs/handbook/decorators.html#parameter-decorators) are supported, not Babel’s;
```
Comment (papb) [inline readme.md:206]: Maybe you can link to Babel decorators as well?

File: readme.md (hunk 1 above)
Comment (sindresorhus) [reply readme.md:205]: `accessors` is the correct term, but I think `getters/setters` is better as it's more commonly known.

File: readme.md (hunk 2 above)
Comment (fregante) [reply readme.md:206]:
```suggestion
- Only [TypeScript’s decorators](https://www.typescriptlang.org/docs/handbook/decorators.html#parameter-decorators) are supported, not [Babel’s](https://babeljs.io/docs/en/babel-plugin-proposal-decorators), which use a different version of the [Decorators Proposal](https://github.com/tc39/proposal-decorators/issues/69);
```

**Pull request merged**